File: src/app/domain/sales-order.ts
import { SearchResultBase } from './search-result-base';

export class Sales_Order extends SearchResultBase {
  id(): string {
    return String(this.SalesOrderID);
  }
  name(): string {
    return this.MaterialName;
  }
  quantity(): number {
    return this.Quantity;
  }
    SalesOrderID: number;
    MaterialName: string;
    Quantity: number;    
  }

File: src/Challenge/Models/Sales_Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Challenge.Models
{
    public class Sales_Order
    {
        public int SalesOrderID { set; get; }
        public string DeliveryDate { set; get; }
        public string Customer { set; get; }
        public string MaterialID { set; get; }
        public string MaterialName { set; get; }
        public int Quantity { set; get; }
        public int TotalValue { set; get; }
    }
}